prometheus/histogram.go

```diff
@@ -205,9 +207,10 @@ func newHistogram(desc *Desc, opts HistogramOpts, labelValues ...string) Histogr
 		}
 	}
 	// Finally we know the final length of h.upperBounds and can make buckets
-	// for both counts:
-	h.counts[0].buckets = make([]uint64, len(h.upperBounds))
-	h.counts[1].buckets = make([]uint64, len(h.upperBounds))
+	// for both counts (with one memory allocation):
+	bothBounds := make([]uint64, 2*len(h.upperBounds))
+	h.counts[0].buckets = bothBounds[:len(bothBounds)/2]
+	h.counts[1].buckets = bothBounds[len(bothBounds)/2:]
```

Comment:
In the same spirit as above, this is not the hot path and this kind of micro optimization makes the code much less readable and any change thereof error-prone, while not providing a measurable gain in return.